File: Algorithms/Searching & Sorting/Counting Sort/Java/CountingSort.java
class CountingSort { 
	
	void sort(int arr[]) { 
		int n = arr.length; 

		int output[] = new int[n]; 

		int count[] = new int[256]; 
		for (int i=0; i<256; ++i) 
			count[i] = 0; 

		for (int i=0; i<n; ++i) 
			++count[arr[i]]; 


		for (int i=1; i<=255; ++i) 
			count[i] += count[i-1]; 

		for (int i = n-1; i>=0; i--) 
		{ 
			output[count[arr[i]]-1] = arr[i]; 
			--count[arr[i]]; 
		} 

		for (int i = 0; i<n; ++i) 
			arr[i] = output[i]; 
	} 

	// Driver method 
	public static void main(String args[]) 
	{ 
		CountingSort ob = new CountingSort(); 
		int arr[] = {10,5,8,4,6,8,7,6,9,23,13,84,94,64,21,84,10,52.57}; 

		ob.sort(arr); 

		System.out.print("Sorted character array is "); 
		for (int i=0; i<arr.length; ++i) 
			System.out.print(arr[i]); 
	} 
} 

File: Algorithms/Searching & Sorting/Counting Sort/Python/counting-sort.py
def counting_sort(array, maxval):
    m = maxval + 1
    count = [0] * m               # init with zeros
    for a in array:
        count[a] += 1             # count occurences
    i = 0
    for a in range(m):            
        for c in range(count[a]):
            array[i] = a
            i += 1
    return (array,count)

print(counting_sort([1, 4, 7, 2, 1, 3, 2, 1, 4, 2, 3, 2, 1], 7))
#                     Sorted Array                  Corresponding Element Count in Array
# Output: [1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 4, 4, 7],      [0, 4, 4, 2, 2, 0, 0, 1]
